hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaOperationUtil.java

```diff
@@ -201,4 +206,26 @@ public static long newContractExpiryIn(MessageFrame frame) {
 
 		return supplierExecution.get();
 	}
+
+	// TODO: Add JavaDocs
+	public static void setOriginalReadValue(
+			final MessageFrame frame,
+			final Address address,
+			final Bytes32 key,
+			final UInt256 storageValue
+	) {
+		// Store the read if it is the first read for the slot/address
+		var updater = frame.getWorldUpdater();
+		while (updater != null && !(updater instanceof HederaWorldState.Updater)) {
```

Comment:
What if we replaced this with `frame.getMessageFrameStack().getLast().parentUpdater()` (or something similar going straight to the root)?  Makes it constant time rather than linear against stack depth.